### PR TITLE
Use a generic QA error object

### PR DESCRIPTION
**Pull request by SilentSpike**
This basically introduces a new `qaError` object to replace the `krError` and `iOsmError` objects.

This object require a `service` property to be initialised so that it can be interacted with generally (if we define the same generic functions in each service).

Additionally if `error_type` is initialised data can be read from `data/qa_errors.json` where I'm currently specifying an `icon` and `category` (only for ImproveOSM at the moment). Category is currently unused, but I'd like to convert error colouring to be based on this for some cross-service consistency.

### CODE REVIEW

File: modules/services/improveOSM.js [hunk 1]
```diff
@@ -210,12 +210,13 @@ export default {
...
 
-                                var d = new iOsmError({
+                                var d = new qaError({
+                                    // Info required for every error
                                     loc: loc,
-                                    comments: null,
-                                    error_subtype: '',
+                                    service: 'improveOSM',
                                     error_type: k,
-                                    icon: 'fas-long-arrow-alt-right',
```
Comment (bhousel) [inline modules/services/improveOSM.js]: Can you add this icon back?  It got missed because I only added it last night  😆

Comment (bhousel) [reply modules/services/improveOSM.js]: or are this in data/qa_errors.json now?  thats ok too..

Comment (SilentSpike) [reply modules/services/improveOSM.js]: Yeah this has been moved to the .json

I figure values we're setting in iD and not those being read from the service API can go there.

**Comment by SilentSpike** [comment]
I'll push a fix for that error and then I'm happy for this to be merged (if you think this is the right way to go, I'm not super experienced with this kind of framework design so there may be a better way).

There's more to do, but can be separately done once this is in place.

**Comment by bhousel** [comment]
Thanks @SilentSpike, I just merged this 👍

**Pull request merged**